### PR TITLE
Features/psp 3131

### PR DESCRIPTION
**Pull request by FuriousLlama**
image

### CODE REVIEW

File: backend/dal/Repositories/ResearchFileRepository.cs [hunk 1]
```diff
@@ -104,12 +106,12 @@ public Paged<PimsResearchFile> GetPage(ResearchFilter filter)
...
 
         /// <summary>
-        /// Generate a new R File in format RFile-XXXXXXXXXX using the research id.
+        /// Generate a new R File in format R-X using the research id.
         /// </summary>
         private string GenerateRFileNumber()
         {
             long nextResearchFileId = this.GetNextResearchSequenceValue();
-            return $"RFile-{nextResearchFileId.ToString().PadLeft(10, '0')}";
+            return $"R-{nextResearchFileId}";
```
Comment (FuriousLlama) [inline backend/dal/Repositories/ResearchFileRepository.cs:114]: R file does not have the file part nor the zero padding anymore

File: backend/dal/Services/ResearchFileService.cs [hunk 2]
```diff
@@ -39,26 +39,24 @@ public PimsResearchFile Add(PimsResearchFile researchFile)
 
             foreach (var researchProperty in researchFile.PimsPropertyResearchFiles)
             {
-                var pid = researchProperty.Property.Pid.Value;
-                try
+                if (researchProperty.Property.Pid.HasValue)
```
Comment (FuriousLlama) [inline backend/dal/Services/ResearchFileService.cs:42]: This was causing exceptions when the pid was null

File: frontend/src/features/mapSideBar/MotiInventoryContainer.tsx [hunk 3]
```diff
@@ -93,12 +118,7 @@ export const MotiInventoryContainer: React.FunctionComponent<IMotiInventoryConta
       showCloseButton
       onClose={props.onClose}
     >
-      <InventoryTabs
-        LtsaView={<LtsaTabView ltsaData={ltsaData} ltsaRequestedOn={ltsaDataRequestedOn} />}
-        PropertyView={
-          showPropertyInfoTab ? <PropertyDetailsTabView property={propertyViewForm} /> : null
-        }
-      />
+      <InventoryTabs tabViews={tabViews} defaultTabKey={defaultTab} />
```
Comment (FuriousLlama) [inline frontend/src/features/mapSideBar/MotiInventoryContainer.tsx:121]: Refactored to allow dynamic tabs to be passed down

File: frontend/src/features/mapSideBar/tabs/InventoryTabs.tsx [hunk 4]
```diff
@@ -2,40 +2,38 @@ import TabView from 'components/common/TabView';
...
-  PropertyView,
-  LtsaView,
+  defaultTabKey,
+  tabViews,
 }) => {
-  const showPropertyInfo = PropertyView !== null;
-
   return (
-    <TabView
-      defaultActiveKey={showPropertyInfo ? InventoryTabNames.property : InventoryTabNames.title}
```
Comment (FuriousLlama) [inline frontend/src/features/mapSideBar/tabs/InventoryTabs.tsx:28]: what tabs to show and which one is the default one is now handled by the parent of this component

File: backend/dal/Services/ResearchFileService.cs [hunk 5]
```diff
@@ -85,5 +83,20 @@ public Paged<PimsResearchFile> GetPage(ResearchFilter filter)
...
+
+        private void PopulateResearchFile(PimsProperty property)
+        {
+            property.PropertyClassificationTypeCode = "CORESTRAT"; // Todo: should be "UNKNOWN"
+            property.PropertyDataSourceEffectiveDate = System.DateTime.Now;
+            property.PropertyDataSourceTypeCode = "PAIMS"; // Todo: should be "PMBC"
+
+            property.PropertyTypeCode = "UNKNOWN";
+
+            property.PropertyStatusTypeCode = "UNSURVYED"; // Todo: should be 'UNKNOWN';
```
Comment (devinleighsmith) [inline backend/dal/Services/ResearchFileService.cs:95]: are these todos being tracked? it seems like we need db changes for the type codes and for the region/district we need a story to get those from the map.

File: frontend/src/features/mapSideBar/tabs/propertyResearch/PropertyResearchTabView.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+          : 'No'
+        : '',
+    legalOpinionObtained:
+      props.researchFile.isLegalOpinionObtained !== undefined
+        ? props.researchFile.isLegalOpinionObtained
+          ? 'Yes'
+          : 'No'
+        : '',
+    documentReference: props.researchFile.documentReference || '',
+    sumaryNotes: props.researchFile.researchSummary || '',
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/tabs/propertyResearch/PropertyResearchTabView.tsx:41]: summary

File: frontend/src/features/mapSideBar/tabs/propertyResearch/PropertyResearchTabView.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    purpose:
+      props.researchFile.propertyPurpose?.map(x => x.propertyPurposeType?.description).join(', ') ||
+      '',
+    legalOpinionRequired:
+      props.researchFile.isLegalOpinionRequired !== undefined
+        ? props.researchFile.isLegalOpinionRequired
+          ? 'Yes'
+          : 'No'
+        : '',
+    legalOpinionObtained:
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/tabs/propertyResearch/PropertyResearchTabView.tsx:34]: can we get a util function for this? we may already have one.

File: frontend/src/features/properties/map/research/add/AddResearchContainer.tsx [hunk 8]
```diff
@@ -46,11 +47,13 @@ export const AddResearchContainer: React.FunctionComponent<IAddResearchContainer
     };
   }, [initialForm, selectedResearchFeature, setSelectedResearchFeature]);
 
-  const saveResearchFile = async (researchFile: Api_ResearchFile, onSuccess: () => void) => {
+  const saveResearchFile = async (researchFile: Api_ResearchFile) => {
     const response = await addResearchFile(researchFile);
+    console.log(response);
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/add/AddResearchContainer.tsx:52]: remove.

File: frontend/src/features/properties/map/research/detail/DetailResearchContainer.tsx [hunk 9]
```diff
@@ -20,6 +23,13 @@ export const DetailResearchContainer: React.FunctionComponent<IDetailResearchCon
 
   const [researchFile, setResearchFile] = useState<Api_ResearchFile | undefined>(undefined);
 
+  const [selectedMenuIndex, setSelectedMenuIndex] = useState<number>(0);
+
+  console.log(researchFile?.researchProperties);
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/detail/DetailResearchContainer.tsx:28]: remove

File: frontend/src/features/properties/map/research/detail/DetailResearchContainer.tsx [hunk 10]
```diff
@@ -20,6 +23,13 @@ export const DetailResearchContainer: React.FunctionComponent<IDetailResearchCon
...
+
+  const menuItems = researchFile?.researchProperties?.map(x => getPropertyName(x)) || [];
+  menuItems.unshift('RFile Summary');
+
   useEffect(() => {
     async function fetchResearchFile() {
       console.log(props.researchFileId);
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/detail/DetailResearchContainer.tsx:35]: remove.

**Comment by devinleighsmith** [comment]
The styling on the navigation bar doesn't seem to match:

image

File: frontend/src/features/properties/map/research/detail/PropertyResearchView.tsx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+import { Api_ResearchFileProperty } from 'models/api/ResearchFile';
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import { pidFormatter } from 'utils';
+
+export interface IPropertyResearchViewProps {
+  researchFileProperty: Api_ResearchFileProperty;
+}
+
+const PropertyResearchView: React.FunctionComponent<IPropertyResearchViewProps> = props => {
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/detail/PropertyResearchView.tsx:26]: Are there tests for this file?

**Review by devinleighsmith — APPROVED** [review]
Mostly just nits, the styling no the navigation bar should get fixed however.

File: frontend/src/features/mapSideBar/tabs/propertyResearch/PropertyResearchTabView.tsx (hunk 7 above)
Comment (FuriousLlama) [reply frontend/src/features/mapSideBar/tabs/propertyResearch/PropertyResearchTabView.tsx:34]: We could, will take a look

File: frontend/src/features/properties/map/research/detail/PropertyResearchView.tsx (hunk 11 above)
Comment (FuriousLlama) [reply frontend/src/features/properties/map/research/detail/PropertyResearchView.tsx:26]: Will be added on a later pr

**Comment by FuriousLlama** [comment]
Future changes and pr comments are done in https://github.com/bcgov/PSP/pull/1380

**Pull request merged**